src/main/java/com/openshift/internal/restclient/okhttp/BasicChallangeHandler.java

```diff
@@ -0,0 +1,42 @@
+/******************************************************************************* 
+ * Copyright (c) 2016 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/
+package com.openshift.internal.restclient.okhttp;
+
+import com.openshift.restclient.authorization.IAuthorizationContext;
+
+import okhttp3.Credentials;
+import okhttp3.Headers;
+import okhttp3.Request.Builder;
+
+/**
+ * 
+ * @author jeff.cantrill
+ *
+ */
+public class BasicChallangeHandler implements IChallangeHandler{
```

Comment:
Challange handlers are only for auth that returns www-authenticate header.  To my knowledge, there is no way for us to progammatically handle oauth; its the same way with 'oc login'  We have a pretty good story now with jbt which allows us to navigate to a browser which is essentially what 'oc login' tells you; go to THIS url to get your token and give it too me.